Financial-Times/n-messaging-client
# Add flags object to guruResult

### Description

Adds [@financial-times/dotcom-ui-flag](https://github.com/Financial-Times/dotcom-page-kit/tree/main/packages/dotcom-ui-flags) package to read the value of flags from `page-kit-flags-embed`
Initialises the flag object on `fetchAsyncConfig` method used for lazy messages. It sends a flatten flag object as a header on the `next-messaging-guru` request, and attaches it to the `guruResult` used on lazy messages
Adds the `page-kit-flags-embed` element to the demo template to mimic FT pages
Changes `embeddedOnboarding` lazy message to use the flags object on the `guruResult` instead of the `embeddedOnboarding` data source

### How to test this code?

- Check out this branch locally and run it with `make demo-watch`
- Go to [FT Toggler](https://toggler.ft.com/) and activate the EmbeddedOnboarding variant on [messageSlotBottom](https://toggler.ft.com/#messageSlotBottom) and variant on [homepageOnboarding](https://toggler.ft.com/#homepageOnboarding)
- Go to the [local demo route](https://local.ft.com:5005/) for n-messaging-client, and you should see the embedded onboarding message. You should finish the journey, and the message should disappear.
- Go back to [FT Toggler](https://toggler.ft.com/) and set the [homepageOnboarding](https://toggler.ft.com/#homepageOnboarding) flag to control. Go back to the [local demo route](https://local.ft.com:5005/) and confirm that the message doesn't show
